### ISSUE TITLE
OS.Windows.resize fails for small sizes

### ISSUE THREAD

**Issue by dkorpel** [vladdeSV/scone#23]
When I resize the console so that the amount of rows is 0, this assert fails:
```
//os.d line 236
assert(SetConsoleScreenBufferSize(consoleOutputHandle, size), "Unable to resize screen buffer");
```
I'm using Windows 10 and the `cmd.exe` terminal.

Looking at the [documentation of SetConsoleScreenBufferSize](https://docs.microsoft.com/en-us/windows/console/setconsolescreenbuffersize):
"The specified dimensions also cannot be less than the minimum size allowed by the system. This minimum depends on the current font size for the console (selected by the user) and the SM_CXMIN and SM_CYMIN values returned by the GetSystemMetrics function."

It doesn't seem to check that currently.

**Comment by vladdeSV** [comment]
Err, that's... an odd thing to do 😅

Just to be clear, you do `window.resize(<width>, 0);` ?

Regardless, I've actually never done that. I'll look into it 👍

**Comment by dkorpel** [comment]
I don't call `window.resize` from my code, I was just testing what happens when you resize the terminal window with the mouse and found that it crashed when I made it too small.

**Comment by vladdeSV** [comment]
Can reproduce, and have identified the culprit as the initial part of the `window.print()`, which does some resizing.

**Comment by vladdeSV** [comment]
And as it turns out it actually is the `os.Windows.resize()` method with fails to resize. (This error can be reproduced with `window.resize(1,1);`.

Changing title: ~~**Program crashes when Window too small**~~ --> **OS.Windows.resize fails for small sizes**

**Comment by vladdeSV** [comment]
To clarify, `window.resize(...)` is called from `window.print()`.

*author's note: that's odd now that i think about it*

**Comment by vladdeSV** [comment]
the console has a minimum size and crashed there. fixed as of b17339fd1c6a8867aa61fac2fb56495964f81216

**Comment by vladdeSV** [comment]
also, sorry for the long delay
